src/client/api/init.c

```diff
@@ -157,11 +157,21 @@ daos_init(void)
 	if (rc != 0)
 		D_GOTO(out_debug, rc);
 
+	/** set up agent */
+	rc = dc_agent_init();
+	if (rc != 0)
+		D_GOTO(out_hhash, rc);
+
+	/** get CaRT configuration */
+	rc = dc_mgmt_net_cfg(NULL);
+	if (rc != 0)
+		D_GOTO(out_obj, rc);
```

Comment:
daos_init() should return -DER not errno. and dc_mgmt_net_cfg could return errno on failure.

Reply:
Fixed.  Thanks.